### PR TITLE
DISPATCH-333: Create new router policies doc

### PR DESCRIPTION
**Pull request by bhardesty**
@ChugR @RoddieKieley here's the updated policies doc. Please review for technical accuracy. Updates include:
* Add new Authorization section to Security chapter (this is the policies content)
* Remove old policies content
* Update policy configuration attribute descriptions

### CODE REVIEW

File: doc/new-book/configuration-security.adoc [hunk 1]
```diff
@@ -412,3 +414,385 @@ listener {
...
+In this example, users in this user group can only connect to the vhost from localhost:
+
+[options="nowrap"]
+----
+remoteHosts: 127.0.0.1, ::1
+----
+====
+
+`sources`::
+A list of source addresses from which users in this group may receive messages. To specify multiple addresses, separate the addresses with either a comma or a space. If you do not specify any addresses, users in this group are not allowed to receive messages from any addresses.
```
Comment (ChugR) [inline doc/new-book/configuration-security.adoc]: Maybe call them 'AMQP source addresses' to signal to the reader that it is not a network address but a protocol thing. Same for targets.

File: doc/new-book/configuration-security.adoc [hunk 2]
```diff
@@ -412,3 +414,356 @@ listener {
...
+[options="nowrap",subs="+quotes"]
+----
+policy = {
+    maxConnections: 10000  // <1>
+    enableVhostPolicy: true  // <2>
+    policyDir: /etc/qpid-dispatch/policies/  // <3>
+    defaultVhost: $default  // <4>
+}
+----
+<1> The maximum number of concurrent client connections allowed for this router. This limit is always enforced, even if no other policy settings have been defined. The limit is applied to all incoming connections regardless of remote host, authenticated user, or targeted vhost. The default value is `65535`.
```
Comment (dmuntima) [inline doc/new-book/configuration-security.adoc]: If possible, please provide the minimum and maximum values for this element.

File: doc/new-book/configuration-security.adoc [hunk 3]
```diff
@@ -412,3 +414,356 @@ listener {
...
+    policyDir: /etc/vhost-policies/ // <1>
+    defaultVhost: $default
+}
+----
+<1> The absolute path to a directory that holds vhost policy definition files in JSON format (*.json). The router processes all of the vhost policies in each JSON file that is in this directory.
+====
+
+. Repeat the previous step for each additional router that should use the vhost policies located in the vhost policy directory.
+
+=== Example: A Vhost Policy Configuration
```
Comment (dmuntima) [inline doc/new-book/configuration-security.adoc]: Suggestion: If this is a heading level three, I think renaming it to "Configuring a Vhost Policy - Example" would be better to maintain parallelism in the headings.

File: doc/new-book/configuration-security.adoc (hunk 2 above)
Comment (ChugR) [reply doc/new-book/configuration-security.adoc]: maxConnections is restricted to be in the range 0..65535 inclusive. That said, a setting of zero works but it renders the router to be useless as no users or management clients can connect to it. The 65535 was chosen to be some limit that is probably not achievable in the real world.

File: doc/new-book/configuration-security.adoc [hunk 4]
```diff
@@ -412,3 +414,356 @@ listener {
...
+=== How {RouterName} Applies Policies
+
+When a client connects to a router, the router determines whether to permit the connection based on the global and vhost policies, and the following properties of the connection:
+
+* The host to which the connection is directed (the vhost)
+* The connection's authenticated user name
+* The host from which the client is connecting (the remote host)
+
+If the connection is permitted, then the router applies a vhost policy that matches the vhost to which the connection is directed. The vhost policy limits are enforced for the lifetime of the connection.
+
```
Comment (ChugR) [inline doc/new-book/configuration-security.adoc]: This description is correct but it glosses over some of the structure within a vhost policy. In a vhost policy  maxConnections, maxConnectionsPerUser, maxConnectionsPerHost, and allowUnknownUser are common for all users. Then based on the user name the vhost policy assigns the remaining policy settings (vhostUserGroupSettings). Users who connect to a given vhost may receive different settings based on what user group the user is assigned.

File: doc/new-book/configuration-security.adoc (hunk 3 above)
Comment (ChugR) [reply doc/new-book/configuration-security.adoc]: +1

File: doc/new-book/configuration-security.adoc (hunk 2 above)
Comment (bhardesty) [reply doc/new-book/configuration-security.adoc]: I updated this to specify that 65535 is both the default and the maximum.

File: doc/new-book/configuration-security.adoc [hunk 5]
```diff
@@ -412,3 +414,356 @@ listener {
...
+* User groups
++
+A user group defines the messaging resources that the group members are permitted to access. Each user group defines the following:
+
+** A set of users that can connect to the vhost (the group members)
+** The remote hosts from which the group members may connect to the router network
+** The AMQP resources that the group members are permitted to access on the vhost
+
+You can configure vhost policies directly in the router configuration file, or create them as JSON files.
```
[the remark this answers is not in the document]
Comment (bhardesty) [reply doc/new-book/configuration-security.adoc]: Done.

File: doc/new-book/configuration-security.adoc (hunk 3 above)
Comment (bhardesty) [reply doc/new-book/configuration-security.adoc]: Good catch. I made this change.

File: doc/new-book/configuration-security.adoc (hunk 4 above)
Comment (bhardesty) [reply doc/new-book/configuration-security.adoc]: I reworked this section to better account for the nuances of vhost policies.

**Pull request closed without merging**